sci-libs/pytorch/pytorch-1.3.1.ebuild

```diff
@@ -0,0 +1,246 @@
+	"${FILESDIR}/0005-Change-library-directory-according-to-CMake-build.patch"
+	"${FILESDIR}/0006-Change-torch_path-part-for-cpp-extensions.patch"
+	"${FILESDIR}/0007-Add-necessary-include-directory-for-ATen-CPU-tests.patch"
+	"${FILESDIR}/0008-Fix-include-directory-variable-of-rocThrust.patch"
+	"${FILESDIR}/0009-Prefer-lib64-for-some-ROCm-packages.patch"
+	"${FILESDIR}/0010-Remove-conversion-ambiguity-in-ternary-operators.patch"
+)
+
+src_prepare() {
+	cmake-utils_src_prepare
+
+	if use rocm; then
+		python "${S}/tools/amd_build/build_amd.py"
+	fi
+}
+
+src_configure() {
+	local blas="Eigen"
+
+	if use atlas; then
+		blas="ATLAS"
+	elif use mkl; then
+		blas="MKL"
+	elif use openblas; then
+		blas="OpenBLAS"
+	fi
+
+	if use rocm; then
+		export HCC_PATH="${HCC_HOME}"
+		export ROCBLAS_PATH="/usr"
+		export ROCFFT_PATH="/usr"
+		export HIPSPARSE_PATH="/usr"
+		export HIPRAND_PATH="/usr"
+		export ROCRAND_PATH="/usr"
+		export MIOPEN_PATH="/usr"
+		export RCCL_PATH="/usr"
+		export ROCPRIM_PATH="/usr"
+		export HIPCUB_PATH="/usr"
+		export ROCTHRUST_PATH="/usr"
+		export ROCTRACER_PATH="/usr"
+	fi
+
+	local mycmakeargs=(
+		-DTORCH_BUILD_VERSION=${PV}
+		-DTORCH_INSTALL_LIB_DIR=lib64
+		-DBUILD_BINARY=$(usex tools ON OFF)
+		-DBUILD_CUSTOM_PROTOBUF=OFF
+		-DBUILD_DOCS=$(usex doc ON OFF)
+		-DBUILD_PYTHON=$(usex python ON OFF)
+		-DBUILD_SHARED_LIBS=$(usex static OFF ON)
+		-DBUILD_TEST=$(usex test ON OFF)
+		-DUSE_ASAN=$(usex asan ON OFF)
+		-DUSE_CUDA=$(usex cuda ON OFF)
+		-DUSE_ROCM=$(usex rocm ON OFF)
+		-DUSE_FBGEMM=$(usex fbgemm ON OFF)
+		-DUSE_FFMPEG=$(usex ffmpeg ON OFF)
+		-DUSE_GFLAGS=$(usex gflags ON OFF)
+		-DUSE_GLOG=$(usex glog ON OFF)
+		-DUSE_LEVELDB=$(usex leveldb ON OFF)
+		-DUSE_LITE_PROTO=OFF
+		-DUSE_LMDB=$(usex lmdb ON OFF)
+		-DCAFFE2_USE_MKL=$(usex mkl ON OFF)
+		-DUSE_MKLDNN=$(usex mkldnn ON OFF)
+		-DUSE_MKLDNN_CBLAS=OFF
+		-DUSE_NCCL=OFF
+		-DUSE_NNPACK=$(usex nnpack ON OFF)
+		-DUSE_NUMPY=$(usex numpy ON OFF)
+		-DUSE_NUMA=$(usex numa ON OFF)
+		-DUSE_OBSERVERS=$(usex observers ON OFF)
+		-DUSE_OPENCL=$(usex opencl ON OFF)
+		-DUSE_OPENCV=$(usex opencv ON OFF)
+		-DUSE_OPENMP=$(usex openmp ON OFF)
+		-DUSE_TBB=OFF
+		-DUSE_PROF=OFF
+		-DUSE_QNNPACK=$(usex qnnpack ON OFF)
+		-DUSE_REDIS=$(usex redis ON OFF)
+		-DUSE_ROCKSDB=OFF
+		-DUSE_ZMQ=$(usex zeromq ON OFF)
+		-DUSE_MPI=$(usex mpi ON OFF)
+		-DUSE_GLOO=$(usex gloo ON OFF)
+		-DBUILD_NAMEDTENSOR=$(usex namedtensor ON OFF)
+		-DBLAS=${blas}
+		-DBUILDING_SYSTEM_WIDE=ON # to remove insecure DT_RUNPATH header
+	)
+
+	cmake-utils_src_configure
+
+	if use python; then
+		distutils-r1_src_configure
+	fi
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	local multilib_failing_files=(
+		libtorch.so
+		libtbb.so
+		libcaffe2_observers.so
+		libc10.so
+		libc10d.a
+		libgloo.a
+		libshm.so
+		libsleef.a
+		libcaffe2_detectron_ops.so
+		libtorch_python.so
+	)
+
+	for file in ${multilib_failing_files[@]}; do
+		mv -f "${D}/usr/lib/$file" "${D}/usr/lib64"
+	done
```

Comment:
This works.  But we'd better fix the build system.

Reply:
Unfortunately, this needs some more experiments and tests, as we kind of go against the upstream here.